lib/jsonld-schemas.js

```diff
@@ -0,0 +1,2157 @@
+    if (process.env.APOS_SEO_DEBUG && imageData && fallbackSource) {
+      console.log(`[SEO] Image fallback used: ${fallbackSource}`);
+    }
+
+    return imageData;
+  }
+
+  // Helper method: Get description with fallback chain
+  getDescriptionWithFallbacks(schemaData, document) {
+    let description = null;
+    let fallbackSource = null;
+
+    // 1. Schema-specific description
+    if (schemaData?.description?.trim()) {
+      description = schemaData.description;
+      fallbackSource = 'schema.description';
+    }
+    // 2. Document seoDescription
+    else if (document?.seoDescription?.trim()) {
+      description = document.seoDescription;
+      fallbackSource = 'document.seoDescription';
+    }
+    // 3. Document excerpt
+    else if (document?.excerpt?.trim()) {
+      description = document.excerpt;
+      fallbackSource = 'document.excerpt';
+    }
+    // 4. Document description
+    else if (document?.description?.trim()) {
+      description = document.description;
+      fallbackSource = 'document.description';
+    }
+
+    if (process.env.APOS_SEO_DEBUG && description && fallbackSource !== 'schema.description') {
+      console.log(`[SEO] Description fallback used: ${fallbackSource}`);
+    }
+
+    return description;
+  }
+
+  // Helper method: Get date with fallback chain
+  getDateWithFallbacks(schemaData, document, dateType = 'published') {
+    let date = null;
+    let fallbackSource = null;
+
+    if (dateType === 'published') {
+      // 1. Schema-specific date
+      if (schemaData?.datePublished) {
+        date = schemaData.datePublished;
+        fallbackSource = 'schema.datePublished';
+      }
+      // 2. Alternative schema field
+      else if (schemaData?.publicationDate) {
+        date = schemaData.publicationDate;
+        fallbackSource = 'schema.publicationDate';
+      }
+      // 3. Document publishedAt
+      else if (document?.publishedAt) {
+        date = document.publishedAt;
+        fallbackSource = 'document.publishedAt';
+      }
+      // 4. Document publicationDate
+      else if (document?.publicationDate) {
+        date = document.publicationDate;
+        fallbackSource = 'document.publicationDate';
+      }
+      // 5. Document datePublished
+      else if (document?.datePublished) {
+        date = document.datePublished;
+        fallbackSource = 'document.datePublished';
+      }
+      // 6. Fall back to createdAt
+      else if (document?.createdAt) {
+        date = document.createdAt;
+        fallbackSource = 'document.createdAt';
+      }
+    } else if (dateType === 'modified') {
+      // For modified dates
+      if (document?.updatedAt) {
+        date = document.updatedAt;
+        fallbackSource = 'document.updatedAt';
+      } else if (document?.createdAt) {
+        date = document.createdAt;
+        fallbackSource = 'document.createdAt';
+      }
+    }
+
+    if (process.env.APOS_SEO_DEBUG && date && fallbackSource && !fallbackSource.startsWith('schema.')) {
```

Comment:
recommend writing a "log" function for this file that does this process.env test so it's not repeated with potential typos.

Reply:
Refactored.